src/darwin/Framework/CHIP/MTRDeviceOverXPC.mm

```diff
@@ -427,6 +427,38 @@ - (void)openCommissioningWindowWithDiscriminator:(NSNumber *)discriminator
     });
 }
 
+- (void)downloadLogOfType:(MTRDiagnosticLogType)type
+                  timeout:(NSTimeInterval)timeout
+                    queue:(dispatch_queue_t)queue
+               completion:(void (^)(NSURL * _Nullable url, NSError * _Nullable error))completion
+{
+    MTR_LOG_DEBUG("Downloading log ...");
+
+    __auto_type workBlock = ^(MTRDeviceControllerXPCProxyHandle * _Nullable handle, NSError * _Nullable error) {
+        if (error != nil) {
+            completion(nil, error);
+            return;
+        }
+
+        [handle.proxy downloadLogWithController:self.controllerID
```

Comment:
How does the XCP client process have access to the file? Is there some magic that happens whenever an NSURL is passed through XCP? Or does it just work in our tests because we happen do be doing it within the same process / sandbox?

Reply:
We’ll solve the sandbox stuff later when we officially support this